src/components/ArticleDetail.jsx

```diff
@@ -0,0 +1,40 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { fetchArticleById } from "../utils/api";
+
+function ArticleDetail() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+
+  useEffect(() => {
+    fetchArticleById(article_id)
+      .then((fetchedArticle) => {
+        setArticle(fetchedArticle);
+      })
+      .catch((error) => {
+        console.error("Error fetching article:", error);
+      });
+  }, [article_id]);
+
+  if (!article) return <p>Loading...</p>;
```

Comment:
What if the article doesn't exist e.g. id=23498 then the loading would hang indefinitely